ui/src/app/components/app-router.tsx

```diff
@@ -38,21 +38,36 @@ const NavRedirect = ({path}) => {
   return null;
 };
 
-export const AppRoute = ({path, data = {}, component: Component}): React.ReactElement => {
+export const AppRoute = ({path, data = {}, guards = [], component: Component}): React.ReactElement => {
   const routeParams = useParams();
   const routeHistory = useHistory();
 
-  return <Route exact={true} path={path} >
-    <Component urlParams={routeParams} routeHistory={routeHistory} routeConfig={data}/>
+  return <Route exact={true} path={path} render={
+    () => {
+      const { redirectPath = null } = fp.find(({allowed}) => !allowed(), guards) || {};
+      return redirectPath
+        ? <NavRedirect path={redirectPath}/>
+        : <Component urlParams={routeParams} routeHistory={routeHistory} routeConfig={data}/>;
+    }}>
   </Route>;
 };
 
 export const ProtectedRoutes = (
   {guards, children}: {guards: Guard[], children: any }): React.ReactElement => {
-  const { redirectPath = null } = fp.find(({allowed}) => !allowed(), guards) || {};
 
-  return redirectPath ? <NavRedirect path={redirectPath}/>
-  : <Fragment>{children}</Fragment>;
+  // Pass the guards to the individual routes. Be sure not to overwrite any existing guards
+  const guardedChildren = fp.flow(
+    fp.flatten,
```

Comment:
What's the purpose of putting `children` in an array, only to immediately flatten them?

Reply:
If a component has a single react child it will not create an array of children - it will be the individual item. If there are multiple children they will already be in an array. I want to ensure we are always dealing with an array of children, I flatten in case children is already an array, and it ends up getting nested.